by-language/python/blob.py

```diff
@@ -0,0 +1,255 @@
+        - https://cratedb.com/docs/crate/reference/en/5.4/general/blobs.html#downloading
+        """
+
+        payload = b""
+
+        for chunk in self.container.get(digest):
+            payload += chunk
+
+        return payload
+
+    def delete(self, digest: str) -> bool:
+        """
+        Delete an item from the BLOB store.
+        """
+        return self.container.delete(digest)
+
+    def provision(self):
+        """
+        Create a new table for storing Binary Large Objects (BLOBs).
+
+        - https://cratedb.com/docs/crate/reference/en/latest/sql/statements/create-blob-table.html
+        """
+        try:
+            self.run_sql(f'CREATE BLOB TABLE "{self.name}";')
+        except ProgrammingError as ex:
+            if "RelationAlreadyExists" not in ex.message:
+                raise
+
+    def refresh(self):
+        """
+        Optionally synchronize data after write operations.
+        """
+        self.run_sql(f'REFRESH TABLE "{self.name}";')
+
+    def run_sql(self, sql: str):
+        """
+        Run SQL statements on the connection.
+        """
+        return self.connection.cursor().execute(sql)
+
+    def __enter__(self):
+        self.connect()
+        return self
+
+    def __exit__(self, *excs):
+        self.disconnect()
+
+
+def setup_logging(level=logging.INFO):
+    """
+    What the function name says.
+    """
+    log_format = "%(asctime)-15s [%(name)-10s] %(levelname)-8s: %(message)s"
+    logging.basicConfig(format=log_format, stream=sys.stderr, level=level)
+
+
+def truncate(value: bytes) -> bytes:
+    """
+    Truncate long string.
+
+    https://stackoverflow.com/a/2873416
+    """
+    ellipsis_ = b"..."
+    maxlength = 100
+    strlength = maxlength - len(ellipsis_)
+    return value[:strlength] + (value[strlength:] and ellipsis_)
+
+
+def example(url: str, container_name: str):
+    """
+    An example conversation with the BLOB store (upload, download, delete).
+    """
+
+    # Define arbitrary content for testing purposes.
+    content = "An example payload.".encode("utf-8")
+    # content = b"\xde\xad\xbe\xef" * 256
+
+    # Upload and re-download content payload.
+    logger.info(f"Uploading: {truncate(content)!r}")
+    with CrateDbBlobContainer(url=url, name=container_name) as container:
+        identifier = container.upload(content)
+        logger.info(f"Identifier: {identifier}")
+
+        downloaded = container.download(identifier)
+        logger.info(f"Downloaded: {truncate(downloaded)!r}")
+
+        deleted = container.delete(identifier)
+        logger.info(f"Deleted: {deleted}")
+
+
+def read_arguments():
+    parser = ArgumentParser()
+    url = parser.add_argument("-u", "--url", type=str)
+    container = parser.add_argument("-c", "--container", type=str)
+
+    actions = parser.add_subparsers(
+        dest="action",
+        title="action",
+        description="valid subcommands",
+        help="additional help",
+    )
```

Comment:
Nit: Define `title`, `description`, and `help` better.

Reply:
do it? ;)